test/cppsuite/test_harness/workload/database_model.h

```diff
@@ -119,6 +147,14 @@ class database {
     }
 
     private:
+    /* Take a const id, not a reference as we're copying in a volatile. */
+    std::string
+    build_collection_name(const uint64_t id)
+    {
+        return std::string("table:collection_" + std::to_string(id));
+    }
+
+    volatile uint64_t _collection_count = 0;
```

Comment:
I don't think `volatile` is going to protect us against torn writes or using the same count twice.

Would prefer to use `std::atomic` or even just incrementing a regular integer while we've acquired the lock since we already have a locking mechanism on this component.

Reply:
We could do the same as _increment_ts in timestamp_manager

Reply:
66de5ba66